phoenix-core/src/main/java/org/apache/phoenix/schema/types/PDouble.java

```diff
@@ -308,4 +309,28 @@ public PhoenixArray newArray(PDataType type, Object[] elements) {
       };
     }
   }
+
+    @Override
+    public int getSign(byte[] bytes, int offset, int length, SortOrder sortOrder,
+            Integer maxLength, Integer scale) {
+        Preconditions.checkNotNull(sortOrder);
+        checkForSufficientLength(bytes, offset, Bytes.SIZEOF_LONG);
+        if (sortOrder == SortOrder.DESC) {
+            bytes =
+                    SortOrder.invert(bytes, offset, new byte[Bytes.SIZEOF_DOUBLE], 0,
+                        Bytes.SIZEOF_DOUBLE);
+            offset = 0;
+        }
```

Comment:
To prevent the creation of a Long, you can use the following and remove your sortOrder logic to get the double here:
    double d = getCodec().decodeDouble(bytes, offset, sortOrder);

Then you can use Math.signum(d) to get the sign for simplification. Note that this code will work for PFloat, PDouble, PUnsignedFloat, and PUnsignedDouble, so it's probably worth having another intermediate class here, something like PRealNumber.